### ISSUE TITLE
witty.er crashes with message "Unhandled Exception: System.FormatException: Input string was not in a correct format."

### ISSUE THREAD

**Issue by SamuelHLewis** [Illumina/witty.er#7]
Hi! I am using witty.er 0.3.3.0 on ubuntu, and I have successfully run several comparisons. However, I am having trouble with one particular truth set VCF, which contains the SV calls for HG00514 from the HGSVC reference set (https://ftp.ncbi.nlm.nih.gov/pub/dbVar/data/Homo_sapiens/by_study/vcf/nstd152.GRCh38.variant_call.vcf.gz). When I run the following command:

`dotnet Wittyer.dll -i testset_first10.vcf -t truthset_first10.vcf -o SV_results`

I get the following error:

```
Unhandled Exception: System.FormatException: Input string was not in a correct format.
   at Ilmn.Das.Core.Tries.Extensions.Try.GetOrThrow[TSource](ITry`1 source)
   at Wittyer.Program.LaunchWittyerMain(String[] args) in /home/sam/bin/witty.er/Wittyer/Program.cs:line 49
   at Wittyer.Program.Main(String[] args) in /home/sam/bin/witty.er/Wittyer/Program.cs:line 22
Aborted (core dumped)
```

The test set VCF works when compared to other truth sets, but the truth set VCF produces the same error when compared against other test sets, so it seems that the problem is with the truth set VCF. This file runs fine through other VCF tools (e.g. bedtools sort), so it appears to be violating a witty.er-specific formatting requirement. I was wondering whether you could see any problem with the format of the truth set VCF (attached), which might be inducing this error?

Thanks!

[truthset_first10.zip](https://github.com/Illumina/witty.er/files/5015027/truthset_first10.zip)

**Comment by Kentalot** [comment]
Yeah, we have pretty strict format requirements to extract the correct information out of the vcf.  Our error messages aren't great though, sorry about that.  Seems like there was a parsing of an int error probably.  But I'll look carefully at this tomorrow.

**Comment by Kentalot** [comment]
Ok so there's two issues with this vcf:
1. CIPOS and CIEND have missing values, which we don't currently support, we could add support by making it simply ignore CIPOS/CIEND, but these entries have partial values (like  `CIPOS=0,.` and `CIEND=.,0` instead of something like `CIPOS=.` or `CIPOS=.,.`, both of which means it was purposefully missing.  The current state looks more like the intention is `CIPOS=0,0` and same with CIEND?  because we can't guess what is intended, we fail instead.  Any feedback would be appreciated.
2. The Format and SAMPLE columns have FORMAT and SAMPLE instead of actual values.  Not sure if you added that to make it pass, but that throws things for a loop since we check those columns for information generally.  I was able to fix this by just changing the FORMAT column for each entry to say GT and the SAMPLE column for each entry to just say .  Here's what I did to get it to work:

[truthset_first10.txt](https://github.com/Illumina/witty.er/files/5018183/truthset_first10.txt)

**Comment by SamuelHLewis** [comment]
Thanks very much for such a quick response! For those issues:

1. The CIPOS and CIEND values came straight from the HGSVC reference file - I guess the mix of partial values might result from this being a consensus set from multiple callers?
2. Yep I added the Format and SAMPLE columns myself as these were missing in the original file, but I hadn't realized that having FORMAT as the value in the Format column would lead to an error

I edited the full truth set VCF to follow the format of your example, and it runs in simple counting mode with no problem :-)

Thanks again for your help!
